wagtail/project_template/Dockerfile

```diff
@@ -1,21 +1,37 @@
 FROM python:3.6
 LABEL maintainer="hello@wagtail.io"
 
-ENV PYTHONUNBUFFERED 1
-ENV DJANGO_ENV dev
+WORKDIR /app
 
-COPY ./requirements.txt /code/requirements.txt
-RUN pip install -r /code/requirements.txt
-RUN pip install gunicorn
+ENV PYTHONUNBUFFERED=1 \
+    # Use production settings as default for the runtime.
+    DJANGO_SETTINGS_MODULE={{ project_name }}.settings.production \
+    # Set the PYTHONPATH so django-admin can be used.
+    PYTHONPATH=/app \
+    # Set Gunicorn's workers to 2 and its port to 8000 as default.
+    WEB_CONCURRENCY=2 \
+    PORT=8000
 
-COPY . /code/
-WORKDIR /code/
+# Expose Gunicorn's port outside of the container.
+EXPOSE 8000
+
+# Install the application server - Gunicorn.
+RUN pip install "gunicorn>=19.8,<19.9"
+
+# Install your application's required dependencies.
+COPY requirements.txt .
+RUN pip install -r requirements.txt
 
-RUN python manage.py migrate
+# Copy your application's files.
+COPY . .
 
+# Don't use the root user.
```

Comment:
isn't it better to do this before adding the application files, this way the layer can't be cached in docker when application files change, or am I wrong?